src/gasPriceOracle/adapters/ethereum.ts

```diff
@@ -3,17 +3,17 @@ import { BigNumber, bnZero } from "../../utils";
 import { GasPriceEstimate } from "../types";
 import { gasPriceError } from "../util";
 
-export async function eip1559(provider: providers.Provider, chainId: number): Promise<GasPriceEstimate> {
-  const feeData = await provider.getFeeData();
+export async function eip1559(provider: providers.Provider, _chainId: number): Promise<GasPriceEstimate> {
+  const [{ baseFeePerGas }, _maxPriorityFeePerGas] = await Promise.all([
+    provider.getBlock("pending"),
+    (provider as providers.JsonRpcProvider).send("eth_maxPriorityFeePerGas", []),
+  ]);
+  const maxPriorityFeePerGas = BigNumber.from(_maxPriorityFeePerGas);
 
-  [feeData.lastBaseFeePerGas, feeData.maxPriorityFeePerGas].forEach((field: BigNumber | null) => {
-    if (!BigNumber.isBigNumber(field) || field.lt(bnZero)) gasPriceError("getFeeData()", chainId, feeData);
-  });
-
-  const maxPriorityFeePerGas = feeData.maxPriorityFeePerGas as BigNumber;
-  const maxFeePerGas = maxPriorityFeePerGas.add(feeData.lastBaseFeePerGas as BigNumber);
-
-  return { maxPriorityFeePerGas, maxFeePerGas };
+  return {
+    maxFeePerGas: maxPriorityFeePerGas.add(baseFeePerGas ?? 0),
```

Comment:
Why can baseFeePerGas ever be undefined or 0?

Reply:
This is just tsc type appeasement, because ethers apparently defines `baseFeePerGas` as optional. We can optionally assert that `baseFeePerGas` is defined; I doubt that would ever throw (though I am querying against blockTag `pending`, so idk whether it might occasionally be undefined in that case).

Reply:
Yeah I'm guessing pending is the issue. If this happens somewhat regularly then a few of 0 would be bad because it would underprice quotes right?

Reply:
If pending is the issue we could always fallback to an older block. However if you see that this is never really undefined we can leave undone for now

Reply:
> Yeah I'm guessing pending is the issue. If this happens somewhat regularly then a few of 0 would be bad because it would underprice quotes right?

I've researched it a bit. I believe it's actually because this field was added with eip1559, so you can still query a pre-1559 block where there would be no `baseFeePerGas`.